File: base_jni/src/main/java/com/mgg/base/trackers/network/NetworkStateTrackerMonitor.java
package com.mgg.base.trackers.network;

import static com.mgg.base.trackers.network.NetworkStateTrackerKt.NetworkStateTracker;

import android.app.Application;

import com.mgg.base.trackers.ConstraintListener;
import com.mgg.base.trackers.ConstraintTracker;

public class NetworkStateTrackerMonitor implements ConstraintListener<NetworkState> {
    private Application mApplication;

    private ConstraintTracker<NetworkState> networkStateTracker;
    public NetworkStateTrackerMonitor(Application application) {
        mApplication = application;
        networkStateTracker = NetworkStateTracker(application, ArchTaskExecutor.getInstance());
    }

    public NetworkState currentState() {
        return networkStateTracker.getState();
    }

    public Boolean isLanConnected() {
        return networkStateTracker.getState().isConnected();
    }

    public Boolean addListener()  {
        networkStateTracker.addListener(this);
        return true;
    }

    public Boolean removeListener() {
        networkStateTracker.removeListener(this);
        return true;
    }

    public void onConstraintChanged(NetworkState newValue) {
        onConstraintChanged(1, newValue.isConnected());
    }

    public native void onConstraintChanged(int connectType, boolean isConnected);
}

File: base_jni/src/main/java/com/mgg/base/trackers/network/NetworkStateTrackerMonitor.kt
package com.mgg.base.trackers.network

import android.app.Application
import androidx.annotation.Keep
import com.mgg.base.trackers.ConstraintListener
import com.mgg.base.trackers.ConstraintTracker
@Keep
class NetworkStateTrackerMonitor constructor(application: Application) : ConstraintListener<NetworkState> {

    /**
     * The tracker used to track network state changes.
     */
    val networkStateTracker: ConstraintTracker<NetworkState> =
        NetworkStateTracker(application.applicationContext, ArchTaskExecutor.getInstance())

    fun currentState(): NetworkState {
        return networkStateTracker.state
    }

    fun isLanConnected(): Boolean {
        return networkStateTracker.state.isConnected
    }

    fun addListener(): Boolean {
        networkStateTracker.addListener(this)
        return true
    }

    fun removeListener(): Boolean {
        networkStateTracker.removeListener(this)
        return true
    }

    override fun onConstraintChanged(newValue: NetworkState) {
        onConstraintChanged(1, newValue.isConnected)
    }

    private external fun onConstraintChanged(connectType: Int, isConnected: Boolean)
}